src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractExtensionMethodImportCompletionProvider.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions.ContextQuery;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers
+{
+    internal abstract class AbstractExtensionMethodImportCompletionProvider : AbstractImportCompletionProvider
+    {
+        protected abstract string GenericSuffix { get; }
+
+        protected override bool ShouldProvideCompletion(Document document, SyntaxContext syntaxContext)
+            => syntaxContext.IsRightOfNameSeparator && IsAddingImportsSupported(document);
+
+        protected async override Task AddCompletionItemsAsync(
+            CompletionContext completionContext,
+            SyntaxContext syntaxContext,
+            HashSet<string> namespaceInScope,
+            bool isExpandedCompletion,
+            CancellationToken cancellationToken)
+        {
+            var syntaxFacts = completionContext.Document.Project.LanguageServices.GetRequiredService<ISyntaxFactsService>();
+            if (TryGetReceiverTypeSymbol(syntaxContext, syntaxFacts, out var receiverTypeSymbol))
+            {
+                var items = await ExtensionMethodImportCompletionService.GetUnimportExtensionMethodsAsync(
+                    completionContext.Document,
+                    completionContext.Position,
+                    receiverTypeSymbol,
+                    namespaceInScope,
+                    isExpandedCompletion,
+                    cancellationToken).ConfigureAwait(false);
+
+                completionContext.AddItems(items.Select(Convert));
+            }
+        }
+
+        private static bool TryGetReceiverTypeSymbol(SyntaxContext syntaxContext, ISyntaxFactsService syntaxFacts, [NotNullWhen(true)] out ITypeSymbol? receiverTypeSymbol)
+        {
+            var parentNode = syntaxContext.TargetToken.Parent;
+            var expressionNode = syntaxFacts.GetLeftSideOfDot(parentNode, allowImplicitTarget: true);
```

Comment:
cute.  do you test for the implicit target case?  that happens in VB iirc.

Reply:
make sure there's a test fo rthis.

Reply:
Extension method is actually only supported for something like 

```vb
        Dim x = New Bar() {.X = .$$}
```

but not 

```vb
        Dim x = New Bar() {.$$}
```

So for simplicity, I decided not to provide completion here at all